### PR TITLE
Fix order of string in editor

### PR DESCRIPTION
**Pull request by stephanlukasczyk**
The string defined in a BibTeX file were shown in random order in the respective GUI editor. While the code already sorts them, by storing the items into a set, it destroys the order again. Fix this behaviour by storing them as a list, which retains the order.

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [X] Tests created for changes (if applicable)
- [X] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [X] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [X] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/test/java/org/jabref/gui/libraryproperties/constants/ConstantsPropertiesViewModelTest.java
Comment (Siedlerchr) [inline src/test/java/org/jabref/gui/libraryproperties/constants/ConstantsPropertiesViewModelTest.java:32]: You don't need to mock the database here, you can simply create a new object without mocking e.g.

`        bibDatabaseContext = new BibDatabaseContext(new BibDatabase());`

File: src/test/java/org/jabref/gui/libraryproperties/constants/ConstantsPropertiesViewModelTest.java
Comment (Siedlerchr) [inline src/test/java/org/jabref/gui/libraryproperties/constants/ConstantsPropertiesViewModelTest.java:40]: You can direclty compare the collections like  this: 
 ```
      var strings  = model.stringsLIstPrroperty.stream().map(ConstantsItemModel::labelProperty).map(StringProperty::getValue).collect(Collectors.toList());
 
Just assertEquals(stringValues, strings)
```

**Comment by Siedlerchr** [comment]
Thanks for the bugfix and for the creation of the test, just some minor "nitpicks" for improving the code style

**Comment by stephanlukasczyk** [comment]
> Thanks for the bugfix and for the creation of the test, just some minor "nitpicks" for improving the code style

Thanks for the quick review, I've attempted to address both suggestions.  Shall I squash the two commits or is it fine to leave them/done automatically during merging?

**Comment by Siedlerchr** [comment]
Just push your new changes, we squash them when merging!

**Review by Siedlerchr — APPROVED** [review]

**Comment by calixtus** [comment]
Codewise looks good, but i mentioned that after adding a new string, the list is probably not sorted again.
Can you check that please?

**Comment by stephanlukasczyk** [comment]
> Codewise looks good, but i mentioned that after adding a new string, the list is probably not sorted again. Can you check that please?

I've just checked, the list is not sorted after adding a new string. This will only be done by hitting the `apply` button and afterwards reopening the dialogue.  However, I would argue that this is an acceptable behaviour here IMHO: if the list would resort as soon as one finishes to add an entry, the new entry might get lost from the immediate focus of the user if one has a larger list of strings. One could, of course, keep the focus on the newly created entry, but that would require probably a larger change in the whole dialogue.

**Comment by calixtus** [comment]
Took a quick look, would be nice if you could quickly implement this: Instead of void return the newly added item from ConstantsPropertiesViewModel#addNewItem and in ConstantsPropertiesView#addString resort after adding the string and use the fxcontrols SelectionModel to select the newly created item.

**Comment by stephanlukasczyk** [comment]
I've implemented the requested reordering @calixtus in 90adb53. However, I did it a bit different than you've suggested: it's now a hook to the `key` column of the table. When adding a new key, the table is resorted and the UI preserves the focus on the newly-created item.  I hope, this fulfills your idea, I am not really into JavaFX.

**Review by calixtus — APPROVED** [review]
LGTM :+1:. The changes need a second review and thus this PR awaits a second review.

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Thanks again for your contribution! Looking forward to more ;)

**Pull request merged**
